### ISSUE TITLE
gridspec.Gridspec doesn't check for consistency in arguments

### ISSUE THREAD

**Issue by goretkin** [matplotlib/matplotlib#5835]
If `width_ratio`s is too small, it fails pretty late too, only when you go to add the subplot.

```python
import matplotlib
import matplotlib.pyplot

gs = matplotlib.gridspec.GridSpec(1,2, width_ratios=[2, 1, 3], height_ratios=[3,4])
plt.gcf().add_subplot(gs[0])
plt.gcf().add_subplot(gs[1])
plt.gcf().savefig("gridspec_issue.png", facecolor="grey")
```

![gridspec_issue](https://cloud.githubusercontent.com/assets/581043/12252517/046ed30a-b8a6-11e5-8860-3bcf8856f199.png)

**Comment by julianvmodesto** [comment]
Working on a pull request for this!

**Comment by jenshnielsen** [comment]
closed by #5892
